bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3StorageFactory.java

```diff
@@ -7,23 +7,29 @@
  *
  *     http://www.apache.org/licenses/LICENSE-2.0
  */
-package io.pravega.segmentstore.storage.impl.extendeds3;
+package io.pravega.storage.extendeds3;
 
 import com.emc.object.s3.S3Config;
 import com.emc.object.s3.jersey.S3JerseyClient;
 import com.google.common.base.Preconditions;
+import io.pravega.common.ConfigSetup;
 import io.pravega.segmentstore.storage.AsyncStorageWrapper;
 import io.pravega.segmentstore.storage.Storage;
 import io.pravega.segmentstore.storage.StorageFactory;
 import io.pravega.segmentstore.storage.rolling.RollingStorage;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.concurrent.GuardedBy;
 
 /**
  * Factory for ExtendedS3 Storage adapters.
  */
 public class ExtendedS3StorageFactory implements StorageFactory {
-    private final ExtendedS3StorageConfig config;
-    private final ExecutorService executor;
+    @GuardedBy("$this")
+    private AtomicReference<ExtendedS3StorageConfig> config;
```

Comment:
Why is an atomic Reference guarded? If it's always guarded you can just use a normal member. Or alternatively If the only reason for the guard is to protect the variable, it can be removed.

Reply:
Fixed.